components/channel/internal/channelhealth/channelHealth.go

```diff
@@ -0,0 +1,73 @@
+package channelhealth
+
+import (
+	"github.com/kyma-incubator/knative-kafka/components/common/pkg/kafka/health"
+	"sync"
+)
+
+// Start The HTTP Server Listening For Requests
+
+type Server struct {
+	health.Server
+
+	// Additional Synchronization Mutexes
+	producerMutex  sync.Mutex	// Synchronizes access to the producerReady flag
+	channelMutex   sync.Mutex	// Synchronizes access to the channelReady flag
+
+	// Additional Internal Flags
+	producerReady  bool         // A flag that the producer sets when it is ready
+	channelReady   bool         // A flag that the channel sets when it is ready
+}
+
+// Creates A New Server With Specified Configuration
+func NewChannelHealthServer(httpPort string) *Server {
+	channelHealth := &Server{}
+	health := health.NewHealthServer(httpPort, channelHealth)
+	channelHealth.Server = *health
+
+	// Return The Server
+	return channelHealth
+}
+
+// Synchronized Function To Set Producer Ready Flag
+func (chs *Server) SetProducerReady(isReady bool) {
+	chs.producerMutex.Lock()
+	chs.producerReady = isReady
+	chs.producerMutex.Unlock()
+}
+
+// Synchronized Function To Set Channel Ready Flag
+func (chs *Server) SetChannelReady(isReady bool) {
+	chs.channelMutex.Lock()
+	chs.channelReady = isReady
+	chs.channelMutex.Unlock()
+}
+
+// Set All Liveness And Readiness Flags To False
+func (chs *Server) ShuttingDown() {
+	chs.Server.ShuttingDown()
```

Comment:
Method names should be verbs, something like "Shutdown" is better than "ShuttingDown"

Reply:
Altered as suggested